### DISCUSSION TITLE
Twitch barely loads with this plugin

### DISCUSSION THREAD

**Discussion by ShiftlessDrifter** [younesaassila/ttv-lol-pro#251] [General]
I've been using this plugin since before it was taken off of the Firefox Add-ons page. I'm using the latest version of Firefox Developer Edition and version 2.1.6 of TTV LOL Pro. At first it was okay, but in the last couple of weeks whenever I go to Twitch it takes forever to load the main page. Then, when I click on a stream it takes several minutes to load. Sometimes I just see a black screen and it will not load at all. I often get the error "Network Error #2000".  It is not my internet connection. When I disable the plugin Twitch loads fast (but I get ads of course). I am using the default Firefox proxy. I have tried using a couple of others I saw people here using but those did not help. What do you suggest I do to fix this problem? Essentially, twitch is completely unusable with this plugin.

**Comment by zGato** [comment]
Twitch is completely unusable with this plugin if you don't use your own infrastructure. The "plugin" (extension) by itself works just fine.
Check the FAQ, has been explained multiple times.

And because you "disable the plugin" and then it instantly loads it doesn't make your internet to not be at fault. My streams don't buffer even if I proxy to Brazil (210ms) or Asia (Tokyo, 280ms) being from Spain, still some US users buffer from connecting to EU servers. That's your ISP's fault for being cheap on paying traffic routing.

TTV LOL PRO is completely independent to TTVLOL devs. V2 is the first version to be on FF store and still up there.

**Reply by ShiftlessDrifter** [reply]
What infrastructure are you talking about? I never needed that before. When I say that disabling the plugin allows Twitch to function, what I mean is the entire domain. I can leave the plugin enabled and go to any website and it loads quickly with no problems. With the plugin enabled, twitch barely loads at all. It is 100% the plugin and not my ISP "being cheap".

What does the FAQ say that is so important? It does mention there is a V1 version of the plugin which I honestly was not aware of. I will see how I can go about loading that. I find this whole manual / roundabout process of getting a browser plugin to work very confusing. All I know is 3-4 weeks ago everything was fine. Now there are problems and there is no obvious solution. I will read the FAQ again and hope that fixes my problems, but I am not terribly optimistic.

**Edit:** Yeah, 1.93 is still giving me ads on twitch, but the platform does load at normal speeds which is nice. I just need to get rid of the ads somehow. It must be a proxy list or something, which is another thing I'm confused about. Keep in mind, this plugin always used to work out of the gate without me needing to do anything. So I will now have to investigate that. Are proxies the "infrastructure" you were referring to?

**Reply by zGato** [reply]
With infrastructure I mean proxy basically. The FAQ mentions buffering, streams appear as offline, etc. That is generic and Twitch not loading is caused by laissez being enabled, and you having bad peering with the proxy. You've never needed that because you haven't bothered to understand how extensions you install work. If you install an AdBlock extension that relies on proxies and it works out of the box, it means you're using someone's proxy infrastructure. 

The plugin allows you to connect to the Twitch server proxying though a proxy server of your choice, if it doesn't load, either the proxy is down or you have a bad connection to it, or some other thing. 1.1k users proxy on peak hours and it works for all of them with the Firefox public proxy. Perhaps my server is not at fault?

You need to understand that Twitch adblocking mechanisms aren't like YouTube ones. It is becoming harder and harder for the average user overtime. If you find it difficult, deal with the ads, get Turbo or spend a little of your time understanding it.

V1 proxying mechanism isn't near the complexity of V2 one. With V1 even if the proxy is down, you can easily fallback to your normal network and you wouldn't even notice, with V2 it is harder to do that due to browsers limitations and coding involved. 

Join the Discord and once I get a bit of free time I can easily answer your questions. I'm not trying to be mean.
